working/macros/feature-specification.md

```diff
@@ -896,6 +916,28 @@ It is a compile-time error if an macro class constructor invoked by a macro
 application has a parameter whose type is not Code (or any subtype of it) or
 one of the aforementioned primitive types (or a nullable type of any of those).
 
+#### Identifier Scope
+
+The following rules apply to any `Identifier` passed as an argument to a macro
+application, whether as a part of a `Code` expression or directly as an
+`Identifier` instance.
+
+The scope of any `Identifier` argument is the same as the scope in which the
+identifier appears in the source code, which is the same as the scope for any
+other type of annotation on a declaration. This means:
+
+* Identifiers in macro application arguments may only refer to static and top
+  level members.
+* They cannot refer to local or instance variables, as those can never be in scope
+  where a macro application appears.
+* Identifiers referring to static class members may be unqualified if the
+  annotation appears on a member of that class.
+
+All identifiers passed to macro constructors must resolve to a real declaration
+by the time macro expansion has completed. They may resolve to generated
+identifiers, including ones generated by the macro they were passed to,
+although that design may be inadvisable.
```

Comment:
How about adding a TODO here pointing to https://github.com/dart-lang/language/issues/2093 to resolve this?

Reply:
In general I left off because I don't think the spec is the best place to be telling people what patterns to use (some other user facing doc for that would be more discoverable). So the spec just explicitly outlines that possibility, making it clear that you _can_ do it in case it affects implementations, even though I don't expect it to.